csrc/device_lower/utils.cpp

```diff
@@ -345,17 +349,36 @@ std::optional<IterDomain*> getMaybeWarpReductionDim(
     return std::nullopt;
   }
 
-  if (reduction_on_xdim->hasPaddingToMultipleOfWarp()) {
-    return std::optional<IterDomain*>(reduction_on_xdim);
-  }
+  // reduction only in xdim.
+  if (!reduction_on_ydim && !reduction_on_zdim) {
+    std::pair<IterDomain*, IterDomain*> reduction_dims =
+        std::make_pair(reduction_on_xdim, nullptr);
+    if (reduction_on_xdim->hasPaddingToMultipleOfWarp()) {
+      return std::optional<std::pair<IterDomain*, IterDomain*>>(reduction_dims);
```

Comment:
nitpick: `return std::make_pair(reduction_on_xdim, nullptr)`

Reply:
revised to
```
      return std::optional<std::pair<IterDomain*, IterDomain*>>(
          std::make_pair(reduction_on_xdim, nullptr));
```
I think we still want to keep this `std::optional` since the function is `getMaybeWarpReductionDim`, otherwise, we need to remove `Maybe` in the function name and make other changes in its callers.

Reply:
No I wasn't suggesting to change the signature. It should automatically convert that to an std::optional. Is there any benefit of having that explicit?

Reply:
got you. I didn't realize it can auto convert to `std::optional`, changed.